.github/workflows/documentation.yaml

```diff
@@ -8,44 +8,59 @@ on:
 permissions: {}
 
 jobs:
+  discover-modules:
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    outputs:
+      modules: ${{ steps.set-matrix.outputs.modules }}
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@ec9f2d5744a09debf3a187a3f4f675c53b671911 # v2.13.0
+        with:
+          egress-policy: audit
+
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          persist-credentials: false
+
+      - name: Discover modules with README.md
+        id: set-matrix
+        run: |
+          # Find all directories under modules/ that contain both README.md and .tf files
+          MODULES=()
+          while IFS= read -r -d '' dir; do
+            # Check if the directory contains both README.md and at least one .tf file
+            if [[ -f "$dir/README.md" ]] && ls "$dir"/*.tf >/dev/null 2>&1; then
```

Comment:
not sure if we need to check for README, as that assumes its in a good state

worried we would miss new modules if they didnt add a README.md

Reply:
should check only for .tf files?